windows/src/engine/keyman32/kmprocess.cpp

```diff
@@ -141,7 +142,88 @@ BOOL ProcessHook()
 			_td->app->QueueDebugInformation(QID_BEGIN_ANSI, NULL, NULL, NULL, NULL, (DWORD_PTR) &keyinfo);
 	}
 
-	ProcessGroup(gp);
+  if (isUsingCoreProcessor) {
+    PWSTR buf;
+    buf = _td->app->ContextBufMax(MAXCONTEXT);
+    km_kbp_context_item *citems = nullptr;
+    ContextItemsFromAppContext(buf, &citems);
+    km_kbp_context_set(km_kbp_state_context(_td->state.lpActiveKBState), citems);
+    km_kbp_context_items_dispose(citems);
+    km_kbp_process_event(_td->state.lpActiveKBState, _td->state.vkey, static_cast<uint16_t>(Globals::get_ShiftState()));
+  }
+  else {
+    ProcessGroup(gp); // TODO: 5442 remove
+  }
+
+  
+  // Process the action items from the core. This actions will modify the windows context (AppContext).
+  // Therefore it is not required to copy the context from the core to the windows context.
+  for (auto act = km_kbp_state_action_items(_td->state.lpActiveKBState, nullptr); act->type != KM_KBP_IT_END; act++) {
```

Comment:
This should be predicated by `isUsingCoreProcessor`.

Can we put this loop into a separate function in a new file, and make each case a separate function also?

Reply:
Yes `kmprocessactions`